src/main/java/core/basesyntax/model/Activity.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class Activity {
+    private static final int ACTIVITY_TYPE_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String SEPARATOR = ",";
+
+    private final Type activityType;
+    private final String fruitName;
+    private final Integer quantity;
+
+    public Activity(String activityLine) {
+        String[] activitySplit = activityLine.split(SEPARATOR);
+        this.activityType = Type.getType(activitySplit[ACTIVITY_TYPE_INDEX]);
+        this.fruitName = activitySplit[FRUIT_NAME_INDEX];
+        this.quantity = Integer.parseInt(activitySplit[QUANTITY_INDEX]);
```

Comment:
Are you sure you need to do this logic in constructor? Maybe you should do this in DataParser